pkg/monitoring/prometheusagent/secret.go

```diff
@@ -19,9 +19,18 @@ func getPrometheusAgentRemoteWriteSecretName(cluster *clusterv1.Cluster) string
 
 // buildRemoteWriteSecret builds the secret that contains the remote write configuration for the Prometheus agent.
 func (pas PrometheusAgentService) buildRemoteWriteSecret(
-	cluster *clusterv1.Cluster, password string) (*corev1.Secret, error) {
+	cluster *clusterv1.Cluster, password string, mimirEnabled bool) (*corev1.Secret, error) {
+	var url string
+	var remoteWriteName string
+
+	if mimirEnabled {
```

Comment:
if you're in this code, mimir is enabled, you don't need that extra check

Reply:
Removed this as well